src/psyclone/tests/test_files/gocean1p0/infrastructure/argument_mod.f90

```diff
@@ -0,0 +1,87 @@
+!-----------------------------------------------------------------------------
+! (c) The copyright relating to this information/data is jointly owned by
+! the Crown, Met Office and NERC 2013.
+! The contribution of STFC in creating this information/data is acknowledged.
+!-----------------------------------------------------------------------------
+
+!> Module containing definitions that enable kernel meta-data to be
+!! written as valid Fortran.
+module argument_mod
+use iso_c_binding
+use global_parameters_mod
+implicit none
+private
+
+enum, bind(c) 
+   ! The following value is valid for any go_arg.
+   enumerator :: GO_READ
+   ! The following values are only valid for fields.
+   enumerator :: GO_WRITE, GO_READWRITE, GO_INC
+   ! The following values are only valid for globals.
+   enumerator :: GO_MIN, GO_MAX, GO_SUM
+end enum
+
+! Fake array to enable us to provide stencil meta-data in the args() array
+integer, public, parameter :: stencil(0:111,0:111,0:111) = 0
```

Comment:
I'm going to be slightly cheeky here and ask that you change to follow Rupert's suggestion (https://github.com/stfc/dl_esm_inf/issues/9) - assuming that compilers other than gfortran are happy with it.

Reply:
Done. This needs the following merge request in dl_esm_info to be done first: https://github.com/stfc/dl_esm_inf/pull/11 to define the new go_stencil type.

Reply:
Thanks - I've just reviewed it.